ds.go

```diff
@@ -148,42 +175,42 @@ func (dsh *ssDSHandler) List(l store.Items, o store.ListOpt) (int, error) {
 		}
 		c := query.OrderByKey{}
 		q.Prefix = "create"
-		q.Filters = []query.Filter{f}
+		q.Filters = append(q.Filters, f)
 		q.Orders = []query.Order{c}
-		listCounter = dsh.getSortedResults(o.Limit, q, l)
+		list = dsh.getSortedResults(o.Limit, q, factory, list)
```

Comment:
This is not a good way of doing it. You don't need the list to pass as argument now. Remove it.

Make the list above a variable and assign the returning list from your function to the above variable.